commands/announce_node.go

```diff
@@ -19,11 +17,14 @@ func checkAnnounceNode(cmd *commandspb.AnnounceNode) Errors {
 
 	if len(cmd.VegaPubKey) == 0 {
 		errs.AddForProperty("announce_node.vega_pub_key", ErrIsRequired)
-	} else {
-		_, err := hex.DecodeString(cmd.VegaPubKey)
-		if err != nil {
-			errs.AddForProperty("announce_node.vega_pub_key", ErrShouldBeHexEncoded)
-		}
+	} else if !IsVegaPubkey(cmd.VegaPubKey) {
+		errs.AddForProperty("announce_node.vega_pub_key", ErrShouldBeAValidVegaPubkey)
+	}
+
+	if len(cmd.Id) == 0 {
+		errs.AddForProperty("announce_node.id", ErrIsRequired)
+	} else if !IsVegaPubkey(cmd.Id) {
+		errs.AddForProperty("announce_node.id", ErrShouldBeAValidVegaPubkey)
```

Comment:
I would make these as separate `if` s as the conditions in order_cancelation.go file like:
`if len(cmd.Id) > 0 && !IsVegaPubKey ... ` for consistency.
It works that way also though

Reply:
I've made it this way just so we can get more "accurate" error really. I don't mind merging them though. any thoughts?